tensorly/decomposition/candecomp_parafac.py

```diff
@@ -247,20 +264,46 @@ def parafac(tensor, rank, n_iter_max=100, init='svd', svd='numpy_svd',\
 
             factors[mode] = factor
 
-        if tol:
-            if sparsity:
-                low_rank_component = kruskal_to_tensor((weights, factors))
-                sparse_component = sparsify_tensor(tensor - low_rank_component, sparsity)
-                
-                unnorml_rec_error = tl.norm(tensor - low_rank_component - sparse_component, 2)
-            else:
-                # ||tensor - rec||^2 = ||tensor||^2 + ||rec||^2 - 2*<tensor, rec>
-                factors_norm = kruskal_norm((weights, factors))
+        # Start line search if requested. Doesn't yet work for fixed modes.
+        if linesearch and iteration % 2 == 0:
+            jump = 1.1
+            total_jump = 0.0
+
+            old_rec_error = error_calc(tensor, weights, factors, sparsity)
+
+            weights_diff = weights - weights_last
+            factors_diff = [factors[ii] - factors_last[ii] for ii in range(tl.ndim(tensor))]
+
+            while True:
+                # Keep stepping while we're successful.
+                new_weights = weights + jump * weights_diff
+                new_factors = [tl.copy(f) for f in factors]
+
+                for ii in modes_list:
+                    new_factors[ii] += factors_diff[ii]
```

Comment:
So if I understand this section right, you don't use the jump variable for the factors update, rather only for the weights. Therefore you basically go several times in the direction given by the ALS update on the factors. This feels a little weird tbh, I would expect this line to be 
```
new_factors[ii] += jump * factors_diff[ii]
```

Reply:
Oops. That's a typo from fixing an mxnet backend error.